src/installer.rs

```diff
@@ -60,91 +63,135 @@ impl Display for LoaderVersion {
 }
 
 pub async fn fetch_minecraft_versions() -> Result<Vec<MinecraftVersion>> {
-    Ok(reqwest::get("https://meta.quiltmc.org/v3/versions/game").await?.json().await?)
+    Ok(get("https://meta.quiltmc.org/v3/versions/game")
+        .await?
+        .json()
+        .await?)
 }
 
 pub async fn fetch_loader_versions() -> Result<Vec<LoaderVersion>> {
-    Ok(reqwest::get("https://meta.quiltmc.org/v3/versions/loader").await?.json().await?)
+    Ok(get("https://meta.quiltmc.org/v3/versions/loader")
+        .await?
+        .json()
+        .await?)
 }
 
-#[derive(Serialize, Deserialize)]
-struct LaunchProfiles {
-    profiles: HashMap<String, serde_json::Value>,
-    settings: serde_json::Value,
-    version: u32
+/// `Deserialize` is not implemented for a reason
+///
+/// DO NOT deserialise `launcher_profiles.json` into this incomplete struct and write it back as it will cause **data loss**
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct Profile {
+    name: String,
+    #[serde(rename = "type")]
+    profile_type: String,
+    created: DateTime<Utc>,
+    last_version_id: String,
+    icon: String,
 }
 
 pub async fn install_client(args: ClientInstallation) -> Result<()> {
     println!("Installing client: {:#?}", args);
 
     // Verify install location
     if !args.install_location.exists() {
-        return Err(anyhow!("Target directory doesn't exist: {:?}", args.install_location));
+        return Err(anyhow!(
+            "Target directory doesn't exist: {:?}",
+            args.install_location
+        ));
     }
 
     // Resolve profile directory
-    let profile_name = format!("quilt-loader-{}-{}", args.loader_version.version, args.minecraft_version.version);
-    let mut profile_dir = args.install_location.clone();
-    profile_dir.push("versions");
-    profile_dir.push(&profile_name);
+    let profile_name = format!(
+        "quilt-loader-{}-{}",
+        args.loader_version.version, args.minecraft_version.version
+    );
+    let profile_dir = args.install_location.join("versions").join(&profile_name);
 
-    // Delete existing profile
     if profile_dir.exists() {
-        std::fs::remove_dir_all(&profile_dir)?;
+        // Delete existing profile
```

Comment:
Hm, the old position for this comment was fine, considering that yeah, it deletes the profile dir if it exists; Maybe "If the profile to be created already exists, delete it" and a move to be above the `if` would work better?

Reply:
I felt that the code explains itself (if profile_dir exists.. yada yada), which is why I removed it.